pandas/tseries/index.py

```diff
@@ -1257,6 +1257,12 @@ def get_loc(self, key, method=None):
                                           'when key is a time object')
             return self.indexer_at_time(key)
 
+        # check if it's a Period and the frequencies are the same - otherwise a monthly period would match for
+        # a daily timestamp at the beginning of the month. NB: 'B' and 'D' therefore won't match
+        if isinstance(key, com.ABCPeriod) and key.freq == self.freq:
```

Comment:
not what I meant. I think these should work IF its a ``Period``. My question is that the freq is irrelevant if *either* the start/end date is in the ``DatetimeIndex``? So give some examples with a freq of ``None`` , but that the ``Period`` can match